internal/weldr/api.go

```diff
@@ -2763,11 +2782,22 @@ func (api *API) composeTypesHandler(writer http.ResponseWriter, request *http.Re
 	}
 
 	// Get the distro specific arch so that we can return the correct list of image types
-	arch, err := d.GetArch(api.archName)
+	archName, err := api.parseArch(request.URL.Query())
```

Comment:
Is parseArch really needed? The only way to tell if the arch name is valid is to make a `d.GetArch` call with it, so I'm not sure any other kind of parsing is needed.